### DISCUSSION TITLE
Best way to have automatic profiles for different displays

### DISCUSSION THREAD

**Discussion by Zabooby** [mpv-player/mpv#14749] [Q&A] [answered]
I have a laptop and a monitor which I switch between when using mpv. They both have the same resolution but the monitor is 100hz while the laptop is 60hz. 

Does mpv automatically detect the change in refresh rate and adjust fps/settings automatically or do I have to put a setting in mpv.conf?

I did find a script called display-name from [here](https://github.com/CogentRedTester/mpv-scripts/?tab=readme-ov-file#display-name) that can allow me  to use conditional auto profiles with the name of the monitors, but I wanted to know if the method above is possible?

**Comment by Andarwinux** [comment]
mpv is per-display aware, but GPU drivers and DWM can have unknowable effects that cause per-display aware is not working as expected, especially cross GPU display.

**Reply by Zabooby** [reply]
The monitor is running off the same laptop, so the gpu is the same.  

What is the extent of mpv's awareness, does it automatically adapt to the higher refresh rate? Is there any need for me to setup a conditional auto profile for the monitor?

**Reply by Andarwinux** [reply, marked as answer]
If your laptop has a dGPU but no BIOS mux switch, it's highly likely that internal display is output by iGPU and external display is output by dGPU.

mpv can sense the refresh rate of the display it is on, so no conditional profile is needed, you can confirm this with stats.

**Reply by Zabooby** [reply]
I do have a dGPU, how do I check which gpu is driving which display?

**Reply by Andarwinux** [reply]
Windows Settings > System > Display > Advanced Graphics Settings

**Reply by Zabooby** [reply]
Ahh yeah I should have guessed that. Thanks for all the help!
